lib/tlm/tlm_interpreter.go

```diff
@@ -3,10 +3,24 @@ package tlm
 import (
 	"encoding/binary"
 	"fmt"
-	"github.com/AarC10/GSW-V2/proc"
 	"math"
+	"strings"
 )
 
+type Measurement struct {
+	Name       string `yaml:"name"`
+	Size       int    `yaml:"size"`
+	Type       string `yaml:"type,omitempty"`
+	Unsigned   bool   `yaml:"unsigned,omitempty"`
+	Endianness string `yaml:"endianness,omitempty"`
+}
+
+type TelemetryPacket struct {
+	Name         string   `yaml:"name"`
+	Port         int      `yaml:"port"`
+	Measurements []string `yaml:"measurements"`
+}
+
 func InterpretUnsignedInteger(data []byte, endianness string) interface{} {
```

Comment:
returning the empty interface is kinda weird here and usually something to avoid. Is there any constraint we can put on the outputs to make them distinguishable from 'literally any type there ever was'. Short of that, could we at least define an alias to interface{} to provide some meaning to the human looking at these functions even if it means nothing to the type system

Reply:
Also, might want to make this return the result and a descriptive error rather than just printing an error. Printing the error is probably not what you want in the long term if you want to provide real context to the issue. One approach might be defining a global
```go
var ErrBadMeasurement = errors.New("failed to convert measurement")
```
and returning that, maybe wrapped with some context kinda following the pattern of https://pkg.go.dev/errors#Is.

Reply:
Theres also these bad boys https://pkg.go.dev/golang.org/x/exp/constraints#Integer

Reply:
The thing you linked is an experimental feature and you cannot return a constrained interface type. Maybe we can look into other options in the future

Reply:
ah cring